pkg/runtime/function/cast_char.go

```diff
@@ -0,0 +1,214 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package function
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"unicode/utf8"
+)
+
+import (
+	"github.com/arana-db/arana/pkg/proto"
+	"github.com/arana-db/arana/pkg/util/runes"
+)
+
+import (
+	gxbig "github.com/dubbogo/gost/math/big"
+	"github.com/pkg/errors"
+	"golang.org/x/text/encoding/charmap"
+	"golang.org/x/text/encoding/simplifiedchinese"
+	"golang.org/x/text/encoding/unicode"
+	"golang.org/x/text/transform"
+)
+
+// FuncCastChar is  https://dev.mysql.com/doc/refman/5.6/en/cast-functions.html#function_cast
+const FuncCastChar = "CAST_CHAR"
+
+var _ proto.Func = (*castcharFunc)(nil)
+
+func init() {
+	proto.RegisterFunc(FuncCastChar, castcharFunc{})
+}
+
+type castcharFunc struct{}
+
+func (a castcharFunc) Apply(ctx context.Context, inputs ...proto.Valuer) (proto.Value, error) {
+	// expr
+	val1, err := inputs[0].Value(ctx)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	if len(inputs) < 3 {
+		return val1, nil
+	}
+
+	// N
+	var num int64
+	val2, err := inputs[1].Value(ctx)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	d2, _ := gxbig.NewDecFromString(fmt.Sprint(val2))
+	if d2.IsNegative() {
+		ret, err := d2.ToInt()
+		if err != nil {
+			return "", err
+		}
+		num = ret
+	} else if !strings.Contains(fmt.Sprint(val2), ".") {
+		ret, err := d2.ToInt()
+		if err != nil {
+			return "", err
+		}
+		num = ret
+	} else {
+		ret, err := d2.ToFloat64()
+		if err != nil {
+			return "", err
+		}
+		num = int64(ret)
+	}
+
+	// charset_info
+	val3, err := inputs[2].Value(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	return a.getResult(runes.ConvertToRune(val1), num, fmt.Sprint(val3))
+}
+
+func (a castcharFunc) NumInput() int {
+	return 3
```

Comment:
Here is the minimal length of arguments, but it seems that actual length can be less than 3, please check it.

Reply:
sql：select cast('hello' as char(2) character set gbk)